src/memory/AmdFlash.hh

```diff
@@ -316,8 +337,29 @@ private:
 		BF8<1> sectorLock;
 		BF8<0> continuity;
 	} statusRegister;
+
+	struct EraseOperation final {
+		size_t index = 0;
+		std::vector<uint8_t> buffer;
+
+		bool isPending(const Sector& sector) const;
+		void markPending(const Sector& sector);
+		void reset();
+		template<typename Archive>
+		void serialize(Archive& ar, unsigned version);
+	} erase;
+
+	struct SyncOperation final : Schedulable {
+		friend class AmdFlash;
+		explicit SyncOperation(Scheduler& s) : Schedulable(s) {}
+		void executeUntil(EmuTime::param time) override {
+			auto& outer = OUTER(AmdFlash, syncOperation);
+			outer.execOperation(time);
+		}
+	} syncOperation;
 };
 SERIALIZE_CLASS_VERSION(AmdFlash, 4);
+SERIALIZE_CLASS_VERSION(AmdFlash::EraseOperation, 1);
```

Comment:
Version=1 is the default. It's fine to state it explicitly, but you may also omit it.